include/rocksdb/db.h

```diff
@@ -110,6 +110,16 @@ struct RangePtr {
   RangePtr(const Slice* s, const Slice* l) : start(s), limit(l) { }
 };
 
+struct IngestExternalFileArg {
```

Comment:
Personally I think making this class immutable doesn't have much value (since the contract is very clear) and it is not something that tends to be modified often. It only makes the API slightly harder to use, as people can't modify the files vector as needed, and is forced to make copies. A simple struct is good enough. Not a huge deal but just want to point it out for your consideration.

Reply:
also for the sake of consistency: the ColumnFamilyHandle * also isn't const.

Reply:
Yeah. The types used to be const reference, then I changed from reference to copy but forgot to remove the const. Will update.

Reply:
Thanks. Also, having the non-default constructor prevents the default constructor. Can we add the default constructor as well?

Reply:
How about removing the non-default constructor? It's member variables are either STL-types, or pointers. Like this
```
struct IngestExternalArg {
  ColumnFamilyHandle* column_family = nullptr;
  std::vector<std::string> external_files;
  IngestExternalFileOptions options;
};
```

Reply:
That'd be great. In this case the non-default ctor provides little value so it can be removed.